lib/puppet/util/windows/user.rb

```diff
@@ -95,8 +101,16 @@ def logon_user(name, password, &block)
     # token has been closed by this point
     true
   end
+
   module_function :logon_user
 
+  def logon_user_by_logon_type(name, password, logon_type, logon_provider, token)
+    LogonUserW(wide_string(name), wide_string('.'), password.nil? ? FFI::Pointer::NULL : wide_string(password), logon_type, logon_provider, token)
+  end
+
+  module_function :logon_user_by_logon_type
```

Comment:
Since this method requires passing constant values for `logon_type` and `logon_provider` that come from elsewhere... and since the only caller of this method is `logon_user`, I would make it a `private_class_method` rather than a publicly exposed `module_function`

Reply:
Done.

Reply:
Looks like you still have `module_function` there. You might have to change the method declaration to `def self.logon_user_by_logon_type` as well for `private_class_method` to work properly.